src/de/received_packet.rs

```diff
@@ -44,14 +68,20 @@ impl<'a> ReceivedPacket<'a> {
             }
         }
 
+        Ok(packet.packet)
+    }
+
+    pub fn from_split_buffer(buf: &'a [u8], tail: &'a [u8]) -> Result<Packet<'a>, ProtocolError> {
+        let mut deserializer = MqttDeserializer::new_split(buf, tail);
```

Comment:
I think this breaks if there's a data chunk spanning the head/tail divide. Perhaps we only want to deserialize the header of the packet, which can't break.